### PR TITLE
Metrics dropdown

### PR DESCRIPTION
**Pull request by pflopez**
Groups and sorts metrics. Metrics in use on other charts show up as disabled.

Screen Shot 2024-04-19 at 13 23 35

### CODE REVIEW

File: src/interface/src/app/plan/project-areas-metrics/chart-data.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ export interface ChartData {
   key: string;
   values: number[];
   metric_layer: string;
+  isPrimary: boolean;
 }
```
Comment (lastminutediorama) [inline src/interface/src/app/plan/project-areas-metrics/chart-data.ts:8]: should we use snake case to match the above?

**Review by lastminutediorama — APPROVED** [review]
lgtm, a couple questions not related to functionality

File: src/interface/src/app/plan/project-areas-metrics/project-areas-metrics.component.html [hunk 2]
```diff
@@ -42,14 +42,43 @@ <h3 class="title">Metrics per Project area</h3>
...
+      <!--          [value]="chart"-->
+      <!--          (selectionChange)="selectDataPoint($event, i)">-->
+      <!--          <mat-option-->
+      <!--            *ngFor="let dataSet of filterData(data, selectedCharts, chart)"-->
+      <!--            [value]="dataSet">-->
+      <!--            {{ dataSet.isPrimary ? '*' : '' }}-->
+      <!--            {{ dataSet.label }}-->
+      <!--          </mat-option>-->
+      <!--        </mat-select>-->
+      <!--      </mat-form-field>-->
```
[the remark this answers is not in the document]
Comment (pflopez) [reply src/interface/src/app/plan/project-areas-metrics/project-areas-metrics.component.html:81]: no, good catch

File: src/interface/src/app/plan/project-areas-metrics/chart-data.ts (hunk 1 above)
Comment (pflopez) [reply src/interface/src/app/plan/project-areas-metrics/chart-data.ts:8]: good call!

**Pull request merged**